ethcore/sync/src/light_sync/mod.rs

```diff
@@ -235,7 +273,8 @@ pub struct LightSync<L: AsLightClient> {
 	pending_reqs: Mutex<HashMap<ReqId, PendingReq>>, // requests from this handler
 	client: Arc<L>,
 	rng: Mutex<OsRng>,
-	state: Mutex<SyncState>,
+	state: Mutex<SyncStateWrapper>,
+	is_idle: Mutex<bool>,
```

Comment:
Maybe it's worth adding a comment, like that:
```suggestion
	// We duplicate this state tracking to avoid deadlocks in `is_major_importing`.
	is_idle: Mutex<bool>,
```
Also I was thinking of using an `std::sync::atomic::AtomicBool` to make sure the deadlock doesn't happen, but maybe it's just me being paranoid.